sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -611,6 +611,289 @@ class SQLContext private[sql](val sparkSession: SparkSession)
     sessionState.catalog.listTables(databaseName).map(_.table).toArray
   }
 
+  ////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////
+  // Deprecated methods
+  ////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * @deprecated As of 1.3.0, replaced by `createDataFrame()`.
+   */
+  @deprecated("Use createDataFrame instead.", "1.3.0")
```

Comment:
Hi, @marmbrus and @srowen .
Do you really need to keep `1.3.0` API?

Reply:
I agree; I also don't feel strongly about it as there isn't much overhead to keeping it, beyond API noise. 
@gatorsmile may have some additional info or data about this.

Reply:
The rubric says nothing about how old the deprecation warning is on purpose. It says we should think about usage.

@gatorsmile do you have any reason to believe these are commonly used functions?

Reply:
At least these Spark books are still using this API in their examples, although the first two claims they are updated using Spark 2.x

- https://www.amazon.com/Mastering-Apache-Spark-2-x-DeepLearning4j/dp/1786462745/ref=sr_1_4?keywords=mastering+Apache+Spark&qid=1584678721&s=books&sr=1-4
- https://www.manning.com/books/spark-in-action
- http://shop.oreilly.com/product/0636920028512.do [first edition]